modules/test/playwright/tests/frontend-data-set-web/customized.spec.ts

```diff
@@ -0,0 +1,534 @@
+		await itemActionButton.click();
+
+		await page
+			.locator(`#${dropdownId}`)
+			.filter({has: page.getByRole('menu')})
+			.waitFor();
+
+		await page
+			.locator(`#${dropdownId}`)
+			.getByRole('menuitem', {
+				exact: true,
+				name: sidePanelActionLabelWithActionTitleContentTitle,
+			})
+			.click();
+
+		await page.getByRole('tabpanel').waitFor();
+
+		const sidePanel = page.getByRole('tabpanel');
+
+		await expect(sidePanel).toBeInViewport();
+
+		await page.getByText(sidePanelActionTitle).waitFor();
+		await expect(page.getByText(sidePanelActionTitle)).toHaveCount(1);
+
+		const iframeElement = await sidePanel.locator('iframe').elementHandle();
+
+		const frame = await iframeElement.contentFrame();
+
+		await expect(
+			frame.locator('.side-panel-iframe-header')
+		).toBeInViewport();
+		await frame.getByText(sidePanelContentTitle).waitFor();
+
+		await expect(frame.getByText(sidePanelContentTitle)).toHaveCount(1);
+
+		await page.keyboard.press('Escape');
+
+		await expect(sidePanel).not.toBeInViewport();
+	});
+
+	await test.step('Side Panel action opens a side panel without title', async () => {
+		const dropdownId = await itemActionButton.getAttribute('aria-controls');
+		await itemActionButton.click();
+
+		await page
+			.locator(`#${dropdownId}`)
+			.filter({has: page.getByRole('menu')})
+			.waitFor();
+
+		await page
+			.locator(`#${dropdownId}`)
+			.getByRole('menuitem', {
+				exact: true,
+				name: sidePanelActionLabelWithoutTitle,
+			})
+			.click();
+
+		await page.getByRole('tabpanel').waitFor();
+
+		const sidePanel = page.getByRole('tabpanel');
+
+		await expect(sidePanel).toBeInViewport();
+
+		await expect(page.locator('.fds-side-panel-title')).toBeInViewport();
+		const panelTitle = await page
+			.locator('.fds-side-panel-title')
+			.allInnerTexts();
+
+		expect(panelTitle).toEqual(['']);
+
+		const iframeElement = await sidePanel.locator('iframe').elementHandle();
+
+		const frame = await iframeElement.contentFrame();
+
+		await expect(
+			frame.locator('.side-panel-iframe-header')
+		).not.toBeInViewport();
+
+		await page.keyboard.press('Escape');
+
+		await expect(sidePanel).not.toBeInViewport();
+	});
+});
+
+test('Use client extensions', async ({page}) => {
+	await test.step('Assert that the cell renderer is invoked and the apple emoji is visible', async () => {
+		const firstColorCell = page
+			.locator('.dnd-tbody > div > div:nth-child(7)')
+			.first();
+
+		await expect(firstColorCell).toContainText('🍏');
+	});
+
+	await test.step('Assert that the filter client extension is working', async () => {
+		const clientExtensionMenuItem = page.getByRole('menuitem', {
+			name: 'Client Extension',
+		});
+
+		const filterButton = page
+			.locator('.filters-dropdown')
+			.getByText('Filter');
+
+		await expect(filterButton).toBeInViewport();
+
+		await clickAndExpectToBeVisible({
+			autoClick: true,
+			target: clientExtensionMenuItem,
+			trigger: filterButton,
+		});
```

Comment:
There was a flaky test here, caused by [missing `await`s on clicks](https://github.com/liferay/liferay-portal/blob/master/modules/test/playwright/tests/frontend-data-set-web/clientExtensionFilter.spec.ts#L69-L84). @kresimir-coko It looks you tried to patch it with `timeout: 500`, but that was not the root cause. In any case, I removed the separate availability check in refactor, as it is not needed. We are checking availability as a part of behavior check.